framework/i18n/MessageFormatter.php

```diff
@@ -341,7 +341,7 @@ private function parseToken($token, $args, $locale)
             case 'selectordinal':
                 throw new NotSupportedException("Message format '$type' is not supported. You have to install PHP intl extension to use this feature.");
             case 'number':
-                if (is_int($arg) && (!isset($token[2]) || trim($token[2]) === 'integer')) {
+                if (is_numeric($arg) && (!isset($token[2]) || trim($token[2]) === 'integer')) {
```

Comment:
I think the condition should be a little more smart.

Possible values of `$token[2]` are `null`, `integer`, `currency`, or `percent`. Currency and percent are not supported, but we should implement a proper fallback for `integer` and `null`.

The following lines should return the same for both intl and fallback:
```php
echo Yii::t('app', 'Here is a big number: {n, number}', ['n' => 2e+8]); // 200,000,000
echo Yii::t('app', 'Here is a big number: {n, number, integer}', ['n' => 2e+8]); // 200,000,000

echo Yii::t('app', 'Here is a big number: {n, number}', ['n' => 200000000.101]); // 200,000,000.101
echo Yii::t('app', 'Here is a big number: {n, number, iinteger}', ['n' => 200000000.101]); // 200,000,000
```